CMakeLists.txt

```diff
@@ -163,6 +163,30 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "mipsel*" OR
   endif()
 endif()
 
+# Set definitions and sources for RISC-V.
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^riscv*")
+  set(PNG_RISCV_VECTOR_POSSIBLE_VALUES check on off)
```

Comment:
I think adding a detect option here makes sense - detect the best way for the current compiler, and it should be the default. If it's not possible, I accept we can't do it. But for users, it won't be easy.